### DISCUSSION TITLE
Run a Script from Right-Click in Finder

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#143] [Share]
https://user-images.githubusercontent.com/36073/114775451-8d52c380-9d2e-11eb-8964-aaca64881203.mp4

**Comment by johnlindquist** [comment]
Note: I imagine it's possible to streamline the Automator step out of this process since the result is a bundle with a couple of XML files. But that's very, very low priority right now :)

**Reply by Antonio-Laguna** [reply]
I've tried this and getting permission errors. Any hints @johnlindquist ?

<img width="372" alt="Screenshot 2021-04-29 at 22 12 24" src="https://user-images.githubusercontent.com/946645/116612556-fb56d780-a937-11eb-9118-2cccb29c25b4.png">

**Reply by Antonio-Laguna** [reply]
I know giving it x permission works but I thought of flagging this since I don't think this was intended ?

**Reply by Antonio-Laguna** [reply]
BTW gave it permission and it just hangs there. No logs anywhere.

**Reply by johnlindquist** [reply]
Looks like it's failing on `~/.kit/script ` permissions. Is that the one you set x permissions to? Also, can you please take a screenshot of your automaton window?

**Reply by Antonio-Laguna** [reply]
I set it to both later on. I tried calling it with `kar` anyways and it just hangs there. I've inspected logs for kit and nothing seems to be there.

**Reply by johnlindquist** [reply]
Do those same commands work from the terminal?

**Reply by Antonio-Laguna** [reply]
No, that's where it hangs too. Sorry, should've been more clear. They're not even called so they're freezing somewhere else.

**Reply by johnlindquist** [reply]
I don't understand what is meant by "they're not even called so their freezing somewhere else".

Any chance you could do a quick screen record of calling them from the terminal?

I'm heading to bed, but I'll dig into this tomorrow

**Reply by Antonio-Laguna** [reply]
Here you go!

Thanks :) 

https://user-images.githubusercontent.com/946645/116649723-41d02480-a980-11eb-8b75-a37faeaedd7e.mp4

**Reply by Antonio-Laguna** [reply]
I've tried reinstalling and it seems to have fixed the issue. Thanks!

**Reply by johnlindquist** [reply]
Cool, glad to hear it's working. 🎉

I would recommend using `~/.kit/script` to launch that script since there is no user input. It will run faster (saves a round-trip to the app) and you can get your console.logs in the terminal.
